src/plopp/plotting/common.py

```diff
@@ -175,18 +176,21 @@ def preprocess(
     return out
 
 
-def preprocess_multi(obj, **kwargs) -> List[sc.DataArray]:
+def inputs_to_nodes(*inputs: PlottableMulti, processor: Callable) -> List[Node]:
     """
-    Pre-process potentially multiple input data for plotting.
-    See :func:`preprocess` for details.
+    Convert a list of inputs or dicts of inputs to a flat list of nodes.
 
     Parameters
     ----------
-    obj:
-        The input objects that will be converted to data arrays.
-    """
-    to_preprocess = from_compatible_lib(obj)
-    if isinstance(to_preprocess, (Mapping, sc.Dataset)):
-        return [preprocess(item, name=name, **kwargs) for name, item in obj.items()]
-    else:
-        return [preprocess(obj, **kwargs)]
+    inputs:
+        The inputs to be converted to nodes.
+    processor:
+        The function that will be applied to each input to convert it to a node.
+    """
+    flat_inputs = []
+    for inp in inputs:
+        if hasattr(inp, 'items') and (not hasattr(inp, 'between')):
```

Comment:
Is this the thing you asked ChatGPT about? Have you considered using something like
```python
def is_pandas_series(x):
    try:
        import pandas as pd
    except ImportError:
        return False
    return isinstance(x, pd.Series)
```
The current check seems too brittle

Reply:
Ah, so if it is a pandas series, then it means you **must** have pandas installed... I guess that should work :+1: